files/default/updatedns

```diff
@@ -42,9 +42,9 @@ function update_dns
 	# Create cluster alias records as applicable
 	#
 	if [[ "${dns_type}" == *"_master"*  ]]; then
-		cli53 rrcreate -i ${dns_type} --failover PRIMARY ${tld} "${alias} AWS ALIAS CNAME ${name} \$self true"
+		cli53 rrcreate -i ${dns_type} --failover PRIMARY --replace ${tld} "${alias} AWS ALIAS CNAME ${name} \$self true"
```

Comment:
have you checkout route53 health check fail over?
https://github.com/barnybug/cli53

Reply:
I have tested this successfully on the CKANTest environment, https://test-dev.data.qld.gov.au/dataset

With a manual failover, there should be no downtime (and I haven't observed any in testing); the servers will simply update the appropriate DNS records to point to themselves. Depending on the exact order of events, there could be a period of time where both the primary and secondary DNS records are pointing to the same server (which could be either the old master or the new one), until it all shakes out, but the primary record should always point to an active server.

With an automatic failover where a server has crashed, there will be downtime of 2-3 minutes until the health check process identifies that the server is no longer responding and selects a new master. Note that the sync process performs an additional health check immediately before syncing, which should reduce the risk of an unhealthy master polluting the replica.

I haven't tried to set up Route53 health checks at this point since we're already using our own application-level health check (via the Solr ping URL).

Reply:
Currently testing an automatic failover, with a hard shutdown of the primary instance. There was an outage for a few minutes, then the other server took over and the site was back up. Still waiting for OpsWorks to recognise that the server is unhealthy.